### PR TITLE
[MM-57963] Log Name and DisplayName of groups

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
The information desired when loggin groups is a bit different then the one wanted in audit logs. This PR adds `Name` and `DisplayName` to the log output and removes `create_at`, `update_at` and `delete_at` as they are hard to read and don't provide much value.

4ea2adbc2b fixes a bug where, for both audit and log entries, `RemoteId` and `MemberCount` were used as pointers. They have to be dereferenced to provide valuable information.

The new log messages look like this:
```json
{
    "timestamp": "2024-04-24 13:30:14.226 +02:00",
    "level": "debug",
    "msg": "Found local group",
    "caller": "ldap/ldap_sync_job.go:543",
    "worker_name": "EnterpriseLdapSync",
    "job_id": "1khg47gnhj85tgkmwxaay7bgcc",
    "job_type": "ldap_sync",
    "job_create_at": "Apr 24 11:29:53.858",
    "group": {
        "allow_reference": false,
        "create_at": 1713786990964
        "delete_at": 0,
        "display_name": "tgroup",
        "has_syncables": false,
        "id": "fo918gdd6pruxg4hfcg6qcqpnc",
        "member_count": 0,
        "name": "",
        "remote_id": "32392d84-74d0-103e-9f75-574fb9e68eef",
        "source": "ldap",
        "update_at": 1713786947991
    }
}
```

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-57963

#### Release Note
```release-note
Log Name and DisplayName of groups
```

### CODE REVIEW

**Review by BenCookie95 — APPROVED** [review]

File: server/public/model/group.go [hunk 1]
```diff
@@ -52,18 +52,30 @@ func (group *Group) Auditable() map[string]interface{} {
...
 
 func (group *Group) LogClone() any {
-	return group.Auditable()
+	return map[string]interface{}{
+		"id":              group.Id,
+		"name":            group.GetName(),
+		"display_name":    group.DisplayName,
+		"source":          group.Source,
+		"remote_id":       SafeDereference(group.RemoteId),
+		"create_at":       MillisToString(group.CreateAt),
```
Comment (wiggin77) [inline server/public/model/group.go:72]: I don't think we should change the format for these standard field names.  Can we instead have different field names for human readable timestamps?

Comment (hanzei) [reply server/public/model/group.go:72]: Should we discuss the time format changes in https://github.com/mattermost/mattermost/pull/26860#discussion_r1585528053 first?

**Comment by hanzei** [comment]
Thanks for all your feedback @wiggin77. I ended up removing the human-readable timestamps as they are too controversial. Let's keep that discussion separate from this PR.

Please take another look. I will update https://github.com/mattermost/mattermost/pull/26860 once this PR is approved.

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:    I recommend updating the sample in the PR description to reflect the output (timestamps).

**Pull request merged**